### DISCUSSION TITLE
✅ 🔗 Faster load time and less bloat by using only necessary CSS/JS

### DISCUSSION THREAD

**Discussion by ZMDx4** [elementor/elementor#14097] [Feature Request]
Hello there, my suggestion is to reduce the bloat caused by both Elementor and Elementor Pro due to the unused CSS/JS and the huge size of HTML DOM while having a modular control would activate only the needed widgets from both Elementor and Elementor Pro.

**Comment by incircolo** [comment]
Yes this is a real problem... i think that it's not possibile to run a website with Elementor and Elementor PRO without a cache plugin which at least reduce the number of server requests.
At least should be very useful to have a JS/CSS concatenation file feature.

**Reply by ZMDx4** [reply]
> Yes this is a real problem... i think that it's not possibile to run a website with Elementor and Elementor PRO without a cache plugin which at least reduce the number of server requests.
> At least should be very useful to have a JS/CSS concatenation file feature.

A real problem once compared to some competitive builders, especially while they make the minumum requests and load necessary files alone. Caching is necessary to serve different type of content at faster speed especially when the targeted visitors are not strictly close from the server location.

**Comment by alexcumbers** [comment]
This is what the latest versions with Asset Load experimental options are addressing.  Give it time to allow these changes to mature.

**Comment by Michaelgimii** [comment]
As they mentioned in the Elementor roadmap video about the new technology in the new version of the editor, I hope that version 2 of the editor will enable the deactivation of unused widgets. Personally, I don't use 60% of Elementor widgets. So I can disable them. This makes the editor much lighter than it is now.

**Comment by Joliamendi** [comment]
Is the main request to add enable/disable widgets feature?

**Comment by nicholaszein** [comment]
Hello!

Thank you for your feedback! 🙏 

Since you submitted this request, Elementor has consistently implemented improvements in performance in different parts of the editor.

The features listed below are a few of the many overall improvements made to the editor, increasing performance and reducing loading time:
- Improved Asset Loading (load only necessary scripts - merged to core)
- Optimized DOM Ouput (remove unnecessary HTML tags - merged to core)
- Improved CSS Loading (load CSS on a per-element basis)
- Inline Font Icons (print icons as SVGs instead of loading fonts)
- Google Fonts Control (enable/disable)
- Google Fonts Load (change font display properties)
- Optimized Image Loading (lazy loading)
- Lazy Load Background Images
- Elements Manager (enable/disable widgets sitewide or on a role-by-role basis)
- Optimized Control Loading (only load content that is needed)

Elementor is committed to improving performance and will continue to do so, in a responsible manner, in the future.

I'm closing the request as **Solved** since it already has been solved by one or multiple features listed above.

Thank you for your contribution!

Kind regards
